src/figuresClasses.ts

```diff
@@ -1,19 +1,94 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.isTriangle() === false) {
+      throw new Error(`Sides ${this.a}, ${this.b}
+        and ${this.c} can't form a triangle`);
+    }
+  }
+
+  isTriangle(): boolean {
+    return (
+      this.a + this.b > this.c
+      && this.a + this.c > this.b
+      && this.b + this.c > this.a
+      && this.a !== 0
+      && this.b !== 0
+      && this.c !== 0
+    );
+  }
 
-export class Circle {
+  getArea(): number {
+    const halfPartOfPerimeter = (this.a + this.b + this.c) / 2;
+    const triangleArea = Math.sqrt(
+      halfPartOfPerimeter * (halfPartOfPerimeter - this.a)
+      * (halfPartOfPerimeter - this.b)
+      * (halfPartOfPerimeter - this.c),
+    );
 
+    return Number(triangleArea.toFixed(2));
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Uncorrect circle radius size');
+    }
+  }
+
+  getArea(): number {
+    const circleArea = this.radius * this.radius * Math.PI;
 
+    return Math.floor(circleArea * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  shape: Shape = 'rectangle';
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (this.isRectangle() === true) {
+      throw new Error('Uncorrect rectangle sides size');
```

Comment:
Incorrect, but which is correct?
For example
`Values should be greater than 0`
You should explain in the errors what exactly are incorrect

Fix in other places

Reply:
✔️